File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/Tèo chạy bộ/MINIGAME2CHAYBO.py
# Nhập dữ liệu từ bàn phím
input_data = input().split()
AB = int(input_data[0])
BC = int(input_data[1])
CA = int(input_data[2])
n = int(input())
S1 = AB + BC + CA
while n > S1:
    n -= S1
if n == AB:
    print("B", end="")
elif n < AB:
    print("AB", end="")
elif n == (AB + BC):
    print("C", end="")
elif n < (AB + BC):
    print("BC", end="")
elif n < (AB + BC + CA):
    print("CA", end="")
elif n == (AB + BC + CA):
    print("A", end="")

File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/Tèo chạy bộ/MINIGAME2CHAYBO.cs
using System;
namespace Teochaybo
{
    public class Program
    {
        static void Main(string[] args)
        {
            string [] input = Console.ReadLine().Split(' ');
            int AB = int.Parse(input[0]);
            int BC = int.Parse(input[1]);
            int CA = int.Parse(input[2]);
            int n = int.Parse(Console.ReadLine());
            int S1 = AB + BC + CA;

            while (n > S1)
                n -= S1;

            if (n == AB)
            {
                Console.Write("B");
                return;
            }
            if (n < AB)
            {
                Console.Write("AB");
                return;
            }
            if (n == (AB + BC))
            {
                Console.Write("C");
                return;
            }
            if (n < (AB + BC))
            {
                Console.Write("BC");
                return;
            }
            if (n < (AB + BC + CA))
            {
                Console.Write("CA");
                return;
            }
            if (n == (AB + BC + CA))
            {
                Console.Write("A");
                return;
            }
        }
    }
}